specs/corla.pvs

```diff
@@ -0,0 +1,923 @@
+%%% model of the Colorado Risk Limiting Audit (CORLA) tool.
+%%% Joe Kiniry
+%%% 11-13 July 2017
+
+%%% This document contains a formal specification of the ColoradoRLA
+%%% system/tool. It is written in a literate style using the PVS
+%%% theorem prover. Writing system specifications in this fashion is
+%%% called (formal) domain engineering. Our formal domain model is
+%%% written in PVS's higher-order logic (HOL). Our informal domain
+%%% model is written in the Free & Fair System Specification Language,
+%%% or FAFESSL for short. FAFESSL is a daughter of Extended BON which
+%%% was, in turn, a daughter of BON, the Business Object Notation.
+
+%%% See our Bibliography project for more information.
+%%% https://github.com/FreeAndFair/Bibliography
+
+%%% In order to relate PVS to FAFESSL, we must define a refinement
+%%% relationship between their two type systems. Informally, that
+%%% mapping is described in the following paragraph.
+
+%%% We map PVS public theories and their contents to FAFESSL constructs.
+%%% The top-level corla theory maps to the FAFESSL system specification,
+%%% theories map to clusters, (PVS) types map to (FAFESSL) types, and
+%%% functions map to features.  We use special comments to denote
+%%% informal specifications which will be extracted using a shell script
+%%% into a well-typed FAFESSL informal domain model specification. 
+
+%%% As mentioned above, we write formal specifications using a
+%%% literate style, a la Knuth. Doing so permits us to document our
+%%% thinking and the system design, from domain modeling and
+%%% engineering all of the way to formal specification and
+%%% verification, in a fashion that produces beautifully printed
+%%% books and interactive hypertext.
+
+%%% In order to write literate PVS we use our old friend Adriaan de
+%%% Groot's scripts available via
+%%% http://www.cs.kun.nl/~adridg/research/PVS-literate.html
```

Comment:
Done. You can review and merge https://github.com/FreeAndFair/ColoradoRLA/pull/35